File: Videos1_20/PythonCode/MainFrame.py
import tkinter as tk
import sys
from Toolbar import Toolbar
from TextPanel import TextPanel
from FormPanel import FormPanel
from StringListener import StringListener
from FormListener import FormListener

class AppStringListener(StringListener):
    def __init__(self, textPanel):
        self.textPanel = textPanel
    def textEmitted(self, text):
        self.textPanel.appendText(text)

class AppFormListener(FormListener):
    def __init__(self, textPanel):
        self.textPanel = textPanel
    def formEventOccurred(self, e):
        name = e.getName()
        occupation = e.getOccupation()
        ageCat = e.getAgeCategory()
        empCat = e.getEmploymentCategory()
        taxId = e.getTaxId()
        usCitizen = e.isUsCitizen()
        self.textPanel.appendText(f"{name}: {occupation}: Age {ageCat}: {empCat}, Citizen: {usCitizen}, Tax ID: {taxId}\n")

class MainFrame(tk.Tk):
    def __init__(self):
        super().__init__()
        self.title("Hello World")
        self.geometry("600x500")

        self.toolbar = Toolbar(self)
        self.textPanel = TextPanel(self)
        self.formPanel = FormPanel(self)

        self.config(menu=self.createMenuBar())

        self.toolbar.setStringListener(AppStringListener(self.textPanel))
        self.formPanel.setFormListener(AppFormListener(self.textPanel))

        self.toolbar.pack(side=tk.TOP, fill=tk.X)
        self.formPanel.pack(side=tk.LEFT, fill=tk.Y)
        self.textPanel.pack(side=tk.LEFT, fill=tk.BOTH, expand=True)

    def createMenuBar(self):
        menuBar = tk.Menu(self)
        
        fileMenu = tk.Menu(menuBar, tearoff=0)
        fileMenu.add_command(label="Exit", command=lambda: sys.exit(0))
        
        windowMenu = tk.Menu(menuBar, tearoff=0)
        showMenu = tk.Menu(windowMenu, tearoff=0)
        
        self.showFormItem = tk.BooleanVar(value=True)
        showMenu.add_checkbutton(label="Person Form", variable=self.showFormItem, command=self.toggleForm)
        
        windowMenu.add_cascade(label="Show", menu=showMenu)
        
        menuBar.add_cascade(label="File", menu=fileMenu)
        menuBar.add_cascade(label="Window", menu=windowMenu)
        
        return menuBar

    def toggleForm(self):
        if self.showFormItem.get():
            self.formPanel.pack(side=tk.LEFT, fill=tk.Y, before=self.textPanel)
        else:
            self.formPanel.pack_forget()

File: Videos1_20/CSharpcode/MainFrame.cs
using System.Windows;
using System.Windows.Controls;

namespace CSharpCode {
    public class MainFrame : Window {
        private TextPanel textPanel;
        private Toolbar toolbar;
        private FormPanel formPanel;

        public MainFrame() {
            Title = "Hello World";
            Width = 600; Height = 500;

            DockPanel dock = new DockPanel();
            toolbar = new Toolbar();
            textPanel = new TextPanel();
            formPanel = new FormPanel();

            Menu menuBar = createMenuBar();
            DockPanel.SetDock(menuBar, Dock.Top); dock.Children.Add(menuBar);
            DockPanel.SetDock(toolbar, Dock.Top); dock.Children.Add(toolbar);
            DockPanel.SetDock(formPanel, Dock.Left); dock.Children.Add(formPanel);
            dock.Children.Add(textPanel);

            toolbar.setStringListener(new CustomStringListener(textPanel));
            formPanel.setFormListener(new CustomFormListener(textPanel));

            Content = dock;
        }

        private Menu createMenuBar() {
            Menu menuBar = new Menu();
            
            MenuItem fileMenu = new MenuItem { Header = "File" };
            MenuItem exitItem = new MenuItem { Header = "Exit" };
            exitItem.Click += (s, e) => Application.Current.Shutdown();
            fileMenu.Items.Add(exitItem);

            MenuItem windowMenu = new MenuItem { Header = "Window" };
            MenuItem showMenu = new MenuItem { Header = "Show" };
            MenuItem showFormItem = new MenuItem { Header = "Person Form", IsCheckable = true, IsChecked = true };
            
            showFormItem.Checked += (s, e) => { formPanel.Visibility = Visibility.Visible; };
            showFormItem.Unchecked += (s, e) => { formPanel.Visibility = Visibility.Collapsed; };
            
            showMenu.Items.Add(showFormItem);
            windowMenu.Items.Add(showMenu);

            menuBar.Items.Add(fileMenu);
            menuBar.Items.Add(windowMenu);
            
            return menuBar;
        }

        // Simulación de las clases anónimas de Java
        private class CustomStringListener : StringListener {
            private TextPanel tp;
            public CustomStringListener(TextPanel tp) { this.tp = tp; }
            public void textEmitted(string text) { tp.appendText(text); }
        }

        private class CustomFormListener : FormListener {
            private TextPanel tp;
            public CustomFormListener(TextPanel tp) { this.tp = tp; }
            public void formEventOccurred(FormEvent e) {
                tp.appendText(e.getName() + ": " + e.getOccupation() + ": Age " + e.getAgeCategory() + ": " + 
                              e.getEmploymentCategory() + ", Citizen: " + e.isUsCitizen() + ", Tax ID: " + e.getTaxId() + "\n");
            }
        }
    }
}